### ISSUE TITLE
Feature request: filter_or

### ISSUE THREAD

**Issue by twolodzko** [tidyverse/dplyr#2758]
It would be grate if dplyr had `filter_or` and `filter_or_` functions as alternatives to `filter` and `filter_`. The `filter` function uses `&` to combine the conditions, while `filter_or` would use `|`. The function is trivial and may seem not needed since one can always use `filter( cond1 | cond2 | cond3 )`, but the feature would enable user to use much less clumsy and dplyr-consistent syntax. Moreover, it would make easer to use in automated scripts that create the code on-the fly.

Example:
`
filter(data, cond1 | cond2 | cond3)`

should give the same results as

`
filter_or(data, cond1, cond2, cond3)`

**Comment by Enchufa2** [comment]
Maybe `filter_any`?

A filter alternative using `xor` to combine the conditions would be very useful too.

**Comment by Enchufa2** [comment]
What about `filter_xor`?

**Comment by twolodzko** [comment]
@romainfrancois many of the functionalities of dplyr can be easily done in base R, so maybe the package is not needed? I have no problem with it not being implemented (I already use my own code to deal with it), it's just a feature request. However the explanation that it can be coded by hand would apply to most of such requests.

**Comment by batpigandme** [comment]
In terms of readability, I'd prefer @Enchufa2's `feature_any()` to `feature_or()`, since the latter vaguely implies that you're filtering _or_ doing something else.

**Comment by romainfrancois** [comment]
@Enchufa2 you could write `filter_xor` using the same principles, e.g. (slightly different): 

```r
filter_xor <- function(.data, ...){
  expr <- reduce(quos(...), function(x, y) expr(xor(!!x, !!y) ) )
  filter( .data, !!expr )
}
```

**Comment by hadley** [comment]
Yeah, there's no reason for this not to go in an extension package.
